### DISCUSSION TITLE
请求支持批量重命名功能

### DISCUSSION THREAD

**Discussion by TingTing1990** [AlistGo/alist#1101] [Ideas]
有时候下载的大量图片和电视剧需要批量重新规则命名，请求支持，感谢！

**Comment by anwen-anyi** [comment]
呃呃呃..你直接本地批量重命名呗QAQ.Alist只是一个列表程序诶~~
还有就是拿阿里云盘来举栗子吧,小白羊开发的客户端都已经被下掉了批量重命名的功能（应该是被官方要求的...）

**Reply by TingTing1990** [reply]
> 呃呃呃..你直接本地批量重命名呗QAQ.Alist只是一个列表程序诶~~ 还有就是拿阿里云盘来举栗子吧,小白羊开发的客户端都已经被下掉了批量重命名的功能（应该是被官方要求的...）

只想PikPak支持😂

**Comment by mm2175** [comment]
配合 Infuse用， 很需要这样的功能

**Reply by DexterYao** [reply]
是的剧集加Season和剧名很需要

**Comment by SakuraYuki56** [comment]
Mac本地连webdav太伤硬盘里，很需要这个功能

**Comment by yanjunhui** [comment]
https://api.aliyundrive.com/v3/file/update
https://open.aliyundrive.com/adrive/v1.0/openFile/update
新的接口是支持单文件重命名操作的.但是目前看不到完整的新接口文档, 不知道是否支持批量重命名.
不过可以循环请求修改的嘛

header 里面 authorization 赋值  access_token

请求 Body 结构

```
{
  "name": "S03E01.mkv",
  "drive_id": "8888888",
  "file_id": "999999999999999999999999",
}
```

**Reply by hankqqu** [reply]
{
  "name": "S03E01.mkv",
  "drive_id": "8888888",
  "file_id": "999999999999999999999999",
}

**Comment by makierika** [comment]
右下菜单中的“正则表达式重命名”可以实现批量重命名

**Reply by thinbug** [reply]
这个正则如何写呢，写了几个感觉都没反应。

**Reply by makierika** [reply]
比如你有一个文件夹“漫长的季节”，里面有13个文件:
The.Long.Season.S01E1.mkv, The.Long.Season.S01E2.mkv, ..., The.Long.Season.S01E13.mkv
你想要批量重命名为:
第01季第1集.mp4, 第01季第2集.mp4, ..., 第01季第13集.mp4
只需要进入该文件夹，点击批量重命名，两个框分别填写：
.\*S(\d{2})E(\d+).\*, 第$1季第$2集.mp4
其中$1,$2分别为括号1和括号2匹配的内容
或者要批量重命名为:
1.mkv, 2.mkv, ..., 13.mkv
则填写:
.\*E(\d+).*, $1.mkv

重点就是括号中匹配的内容可以用$n表示
